r/R/dplyr-funcs-datetime.R

```diff
@@ -351,6 +351,25 @@ register_bindings_duration <- function() {
     delta <- delta$cast(int64())
     start + delta$cast(duration("s"))
   })
+    register_binding("make_difftime", function(num = NULL,
+                                             units = "secs",
+                                             ...) {
+    if (units != "secs") {
+      abort("`make_difftime()` with units other than 'secs' not supported in Arrow")
+    }
+
+    chunks <- list(...)
+
+    if (is.null(num)) {
+      duration <- duration_from_chunks(chunks)
+    } else if (length(chunks) == 0) {
+      duration <- num
+    } else {
+      abort("`make_difftime()` with both `num` and `...` not supported in Arrow")
+    }
```

Comment:
This gets the behavior we want, though the conditions on the if/else make it a little bit funny to follow: If `num` is `NULL`, then do something with `chunks`, if `chunks` don't exist do something with `NULL`.

What if instead we were a bit more explicit and did:

```
if (!is.null(num) && length(chunks > 0)) {
   abort("`make_difftime()` with both `num` and `...` not supported in Arrow")
}

if (!is.null(num)) {
  duration <- num
} else {
  duration <- duration_from_chunks(chunks)
}
```

or flip the last if/else around to do the chunks first.